src/Microsoft.DotNet.MSBuildSdkResolver/StringExtensions.cs

```diff
@@ -0,0 +1,15 @@
+namespace Microsoft.DotNet.MSBuildSdkResolver
+{
+    internal static class StringExtensions
+    {
+        public static int FindFirstNotOf(this string s, string chars, int startIndex)
```

Comment:
Side note: Extension can also be use as a way of writing function chain. But I am still skeptical to write functional code in C#

Let me dig some old argument
In F# you can do things like this, (not real F#)
```F#
let itemsToReport = newLineItems
|> List.filter( fun (l) -> !l.IsPrePaid() || fun(l) -> l.IsProratedCharge())
|> Aggregate
|> RemoveReportedChargesForThisOrder
```

You can do similar stuff by extension method
```C#
var itemsToReport = newLineItems
.Where(l => !l.IsPrePaid() || l.IsProratedCharge())
.Aggregate()
.RemoveReportedChargesForThisOrder()
```